bliss/encoder/variational_dist.py

```diff
@@ -40,6 +43,37 @@ def compute_nll(self, x_cat, true_tile_cat):
         return sum(qk.compute_nll(params, true_tile_cat) for qk, params in fp_pairs)
 
 
+class NllGating(ABC):
+    @classmethod
+    @abstractmethod
+    def get_nll_gating(cls, true_tile_cat: TileCatalog):
+        """Get Gating for nll."""
+
+
+class NullGating(NllGating):
+    @classmethod
+    def get_nll_gating(cls, true_tile_cat: TileCatalog):
```

Comment:
Maybe `__call__` would be a better method name? That's the name of a special method that gets invoked when you treat the object as if it were a function.

Reply:
fixed